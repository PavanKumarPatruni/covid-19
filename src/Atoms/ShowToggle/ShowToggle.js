import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

const ArrowDown = forwardRef(({ color, size, show }, ref) => {
  return (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill={color}
      stroke={color}
      transform={show ? '' : 'rotate(-90)'}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <polyline points="6 9 12 15 18 9"></polyline>
    </svg>
  );
});

ArrowDown.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  show: PropTypes.bool,
};

ArrowDown.defaultProps = {
  size: 24,
  color: '#000000',
  show: false,
};

ArrowDown.displayName = 'ArrowUp';

export default ArrowDown;
