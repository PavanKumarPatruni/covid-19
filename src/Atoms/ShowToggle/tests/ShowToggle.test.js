import React from 'react';
import { shallow, mount } from 'enzyme';

import ShowToggle from '../ShowToggle';

describe('ShowToggle component', () => {
  const props = {
    color: '',
    size: 24,
    show: true,
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<ShowToggle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match snapshot correctly', () => {
    const wrapper = mount(<ShowToggle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    props.show = false;
    const wrapper = shallow(<ShowToggle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match snapshot correctly', () => {
    props.show = false;
    const wrapper = mount(<ShowToggle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
