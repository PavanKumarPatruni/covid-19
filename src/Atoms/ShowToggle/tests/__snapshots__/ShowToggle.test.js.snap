// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowToggle component match snapshot correctly 1`] = `
<svg
  fill=""
  height={24}
  stroke=""
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  transform=""
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <polyline
    points="6 9 12 15 18 9"
  />
</svg>
`;

exports[`ShowToggle component match snapshot correctly 2`] = `
<ArrowUp
  color=""
  show={true}
  size={24}
>
  <svg
    fill=""
    height={24}
    stroke=""
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    transform=""
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="6 9 12 15 18 9"
    />
  </svg>
</ArrowUp>
`;

exports[`ShowToggle component match snapshot correctly 3`] = `
<svg
  fill=""
  height={24}
  stroke=""
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  transform="rotate(-90)"
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <polyline
    points="6 9 12 15 18 9"
  />
</svg>
`;

exports[`ShowToggle component match snapshot correctly 4`] = `
<ArrowUp
  color=""
  show={false}
  size={24}
>
  <svg
    fill=""
    height={24}
    stroke=""
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    transform="rotate(-90)"
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="6 9 12 15 18 9"
    />
  </svg>
</ArrowUp>
`;
